example/helper.rb

```diff
@@ -17,3 +17,25 @@ def info(msg)
     puts "[Stream #{@id}]: #{msg}"
   end
 end
+
+def send_promises(stream)
+  push_streams = []
+
+  # send 10 promises
+  10.times do |i|
+    sleep 1
+    puts 'sending push'
+
+    head = {
+      ':status' => '200',
+      ':path'   => "/other_resource/#{i}",
+      'content-type' => 'text/plain',
+    }
+
+    stream.promise(head) do |push|
+      push.headers(head)
```

Comment:
I guess this one is a bit subtle.. We may want to rethink the API we provide here in the future, but for time being:

http://httpwg.org/specs/rfc7540.html#PushResources

> The PUSH_PROMISE frame includes a header block that contains a complete set of **request header fields** that the server attributes to the request.
> 
> The header fields in PUSH_PROMISE and any subsequent CONTINUATION frames MUST be a valid and complete set of request header fields (Section 8.1.2.3). The server MUST include a method in the :method pseudo-header field that is safe and cacheable.
>
> The server MUST include a value in the :authority pseudo-header field for which the server is authoritative (see Section 10.1). A client MUST treat a PUSH_PROMISE for which the server is not authoritative as a stream error (Section 5.4.2) of type PROTOCOL_ERROR.

So, what we need to do here is:

```ruby
stream.promise({':method': 'GET', ':authority': 'localhost', ':path': '/resource/...'}) do |push|
  push.headers({':status': 200, 'content-length': ..., ... })
  push.data(...)
end
```

Hope that makes sense!